src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -25,7 +25,7 @@ public class JwtPayload : Dictionary<string, object>
     {
         internal const string ClassName = "System.IdentityModel.Tokens.Jwt.JwtPayload";
 
-        internal List<string> _audiences;
+        internal HashSet<string> _audiences;
```

Comment:
Why are we changing to HashSet?

Reply:
It was for automatic deduplication, but since we decided not to worry about duplicates, I'll be changing it

Reply:
changed back to list

Reply:
As mentioned in Teams chat, deduplication is the existing behavior in the JsonWebTokenHandler, if we don't de-duplicate here the behavior for the JwtSecurityTokenHandler will be different than the JsonWebTokenHandler, so we may want to keep the HashSet.